### PR TITLE
blacklist documents to avoid publishing projects #158480658

### PR DESCRIPTION
**Pull request by grpse**
Adds a document to blacklist document base and don't allow user to continue to projects publishing

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -179,10 +179,23 @@ def address_fields_validation
   end
 
   def owner_document_validation
-    if cpf.present? && (published_projects.present? || contributed_projects.present? || publishing_project)
-      unless account_type != 'pf' ? CNPJ.valid?(cpf) : CPF.valid?(cpf)
-        errors.add(:cpf, :invalid)
-      end
+    is_blacklisted = false
+    strippedCpf = ""
+<<<<<<< HEAD
```
Comment (vicnicius) [inline app/models/user.rb]: some merge conflict remainings here.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -137,6 +137,7 @@ def update
   def ban
     authorize resource
     resource.update_column(:banned_at, DateTime.now)
+    BlacklistDocument.create(number: @user.cpf) unless @user.cpf.nil?
```
Comment (devton) [inline app/controllers/users_controller.rb]: we need to strip where too

File: app/models/user.rb [hunk 3]
```diff
@@ -179,10 +179,20 @@ def address_fields_validation
...
-        errors.add(:cpf, :invalid)
-      end
+    is_blacklisted = false
+    strippedCpf = ""
+    
+    if cpf.present?
+      strippedCpf = cpf.gsub(".","").gsub("-","")
+      blentry = BlacklistDocument.find_by_number(strippedCpf)
+      is_blacklisted = !blentry.nil?
+    end
```
Comment (devton) [inline app/models/user.rb]: add some specs for this logic

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -37,6 +38,10 @@
     it { is_expected.not_to allow_value('foo').for(:email) }
     it { is_expected.not_to allow_value('foo@bar').for(:email) }
     it { is_expected.to validate_uniqueness_of(:email) }
+    it "Should be a valid cpf" do
```
Comment (devton) [inline spec/models/user_spec.rb]: what this test test?

**Pull request merged**